packages/stitch/tests/issue-3371.test.ts

```diff
@@ -0,0 +1,151 @@
+import { graphql } from 'graphql';
+
+import { makeExecutableSchema } from '@graphql-tools/schema';
+
+import { stitchSchemas } from '../src/stitchSchemas';
+
+const tags = [
+  { tagId: 't1', label: 'One', creatorId: 'u0', userId: 'u1' },
+  { tagId: 't2', label: 'Two', creatorId: 'u0', userId: 'u1' },
+  { tagId: 't3', label: 'Three', creatorId: 'u0', userId: 'u2' },
+];
+
+const users = [
+  { userId: 'u0', email: 'zero@meh', name: 'User 0' },
+  { userId: 'u1', email: 'one@meh', name: 'User 1' },
+  { userId: 'u2', email: 'two@meh', name: 'User 2' },
+];
+
+const tagSchema = makeExecutableSchema({
+  typeDefs: /* GraphQL */ `
+    type PlatformUser {
+      userId: ID!
+    }
+
+    type User {
+      userId: ID!
+    }
+
+    type UserTag {
+      tagId: ID!
+      label: String!
+
+      creator: PlatformUser
```

Comment:
If I change from `PlatformUser` to `User` then the stitching will work as expected and the test will pass